### ISSUE TITLE
[Feature request] Multi-source downloading

### ISSUE THREAD

**Issue by jrmoserbaltimore** [picotorrent/picotorrent#1184]
It would be nice if PicoTorrent could take a torrent and also a download URL, and download the various blocks from both sources.  I believe BT provides a hash for each block, so you could quickly obtain hashes and download blocks with HTTP resume requests.

**Comment by vktr** [comment]
Would a way of adding web seeds be enough? I think that is basically what you are describing...

**Comment by jrmoserbaltimore** [comment]
Yes, that would work.  I don't know much about BT but I know torrents sometimes come from a site with a direct download, and sometimes spend hours looking for 2 peers when the torrent is old.  I ran into an issue where the Web site download was like 60k/s and the download on BT was 100k/s, so either slow or slower.  Whatever the solution is to this works for me.

**Comment by iamalexei** [comment]
@vktr please, add downloading from web seeds.
